routing/posts/postRoutes.js

```diff
@@ -0,0 +1,94 @@
+/* --- Routes for '/api/posts' --- */
+const express = require('express'); // import the express package
+
+const db = require('../../data/db.js'); // import data helpers
+
+const router = express.Router(); // creates the route
+
+
+// POST - Creates a post using info sent in request body
+router.post('/', (req, res) => {
+  const { title, contents } = req.body;
+  const post =  { title, contents }
+  if (!title || !contents) {
+    res.status(400).json({ error: 'Please provide title and contents for the post.' })
+  }
+  db
+  .insert(post)
+  .then(postID => {
+    res.status(201).json(postID);
+  })
+  .catch(err => {
+    res.status(500).json({ error: 'There was an error while saving the post to the database.' });
+  })
+})
+
+// GET - Returns an array of all posts in database
+router.get('/', (req, res) => {
+  db
+  .find()
+  .then(posts => {
+    res.status(200).json(posts);
+  })
+  .catch(err => {
+    res.status(500).json({ error: 'The posts information could not be retrieved.' });
+  })
+})
+
+// GET - Returns post with the specified ID
+router.get('/:id', (req, res) => {
+  const { id } = req.params;
+  db
+  .findById(id)
+  .then(post => {
+    if (post.length === 0) {
+      res.status(404).json({ error: 'The post with the specified ID does not exist.'});
+      return;
+    }
```

Comment:
I would like to see an else here even if it's not strictly necessary for two reasons:
1) It's more semantic.
2) There's no question that you only go into the second statement if the first statement doesn't evaluate to true..